R-package/tests/testthat/test_lgb.Booster.R

```diff
@@ -174,7 +174,7 @@ test_that("Loading a Booster from a text file works", {
         , bagging_freq = 1L
         , boost_from_average = FALSE
         , categorical_feature = c(1L, 2L)
-        , interaction_constraints = list(c(1L, 2L), 1L)
+        , interaction_constraints = list(1L:2L, 3L, 4L:ncol(train$data))
```

Comment:
This PR doesn't currently have a test confirming that the new automatically-added grouped is actually added to the model.

The change here on this line is passing that group through `params` from the beginning, and the new test added in `test_utils.R` only checks that `.check_interaction_constraints()` adds that group to whatever's passed into it... not necessarily that the output of that function actually makes it all the way to LightGBM's C++ code and therefore affects training.

Could you please add a test that works roughly like the following?

* passes something like `interaction_constraints = list(3L, 1:2L)` through `params` to `lgb.train()`
  - *passing them out of order like that also might help catch some issues*
* checks that the `interaction_constraints` in `$get_params()` of that `Booster` produced looks correct (i.e. includes the automatically-added group)
* checks that `interaction_constraints` parameter looks correct in the model string produced by `$save_model_to_string()`

That'd help improve our confidence that this is working end-to-end.

Reply:
Good idea. I have hard-coded the expected result into an additional unit test that checks both integer and string columns.